internal/binutils/binutils_test.go

```diff
@@ -401,3 +414,125 @@ func TestOpenMalformedMachO(t *testing.T) {
 		t.Errorf("Open: got %v, want error containing 'Mach-O'", err)
 	}
 }
+
+func TestObjdumpVersionChecks(t *testing.T) {
+	// Test that the objdump version strings are parsed properly.
+	type objdumpTest struct {
+		desc string
+		os   string
+		ver  string
+		want bool
+	}
+
+	for _, tc := range []objdumpTest{
+		{
+			// Valid Apple LLVM version string with usable version.
+			desc: "Valid Apple LLVM version string",
+			os:   "darwin",
+			ver: `Apple LLVM version 11.0.3 (clang-1103.0.32.62)
+			Optimized build.`,
+			want: true,
+		},
+		{
+			// Valid Apple LLVM version string with unusable version.
+			desc: "Unusable Apple LLVM version",
+			os:   "darwin",
+			ver: `Apple LLVM version 10.0.0 (clang-1000.11.45.5)
+			Optimized build.`,
+			want: false,
+		},
+		{
+			// Invalid Apple LLVM version string with usable version.
+			desc: "Invalid Apple LLVM version string",
+			os:   "darwin",
+			ver: `Apple LLVM versions 11.0.3 (clang-1103.0.32.62)
+			Optimized build.`,
+			want: false,
+		},
+		{
+			// Valid LLVM version string with usable version.
+			desc: "Valid LLVM version string",
+			os:   "linux",
+			ver: `LLVM (http://llvm.org/):
+			LLVM version 9.0.1
+	
+			Optimized build.`,
+			want: true,
+		},
+		{
+			// Valid LLVM version string with unusable version.
+			desc: "Unusable LLVM version",
+			os:   "linux",
+			ver: `LLVM (http://llvm.org/):
+			LLVM version 6.0.1
+	
+			Optimized build.`,
+			want: false,
+		},
+		{
+			// Invalid LLVM version string with usable version.
+			desc: "Invalid LLVM version string",
+			os:   "linux",
+			ver: `LLVM (http://llvm.org/):
+			LLVM versions 9.0.1
+	
+			Optimized build.`,
+			want: false,
+		},
+		{
+			// LLVM objdump version string contains trunk
+			desc: "Valid LLVM objdump version string with trunk",
+			os:   runtime.GOOS,
+			ver: `LLVM (http://llvm.org/):
+			LLVM version custom-trunk 124ffeb592a00bfe
+			Optimized build.`,
+			want: true,
+		},
+		{
+			// Invalid LLVM objdump version string
+			desc: "Invalid LLVM objdump version string with trunk",
+			os:   runtime.GOOS,
+			ver: `LLVM (http://llvm.org/):
+			LLVM version custom-trank 124ffeb592a00bfe
+			Optimized build.`,
+			want: false,
+		},
+		{
+			// Invalid LLVM objdump version string
+			desc: "Invalid LLVM objdump version string with trunk",
+			os:   runtime.GOOS,
+			ver: `LLVM (http://llvm.org/):
+				llvm version custom-trunk 124ffeb592a00bfe
+				Optimized build.`,
+			want: false,
+		},
+	} {
+		if runtime.GOOS == tc.os {
+			if got := isLLVMObjdump(tc.ver); got != tc.want {
+				t.Errorf("%v: got %v, want %v", tc.desc, got, tc.want)
+			}
+		}
+	}
+	for _, tc := range []objdumpTest{
+		{
+			// Valid GNU objdump version string.
+			desc: "Valid GNU objdump version string",
+			os:   runtime.GOOS,
```

Comment:
Looks like this field isn't used below, so no need to set it?

Reply:
Done.